src/extension/importer/core/src/main/java/org/geoserver/importer/transform/AbstractCommandLineTransform.java

```diff
@@ -0,0 +1,334 @@
+/* (c) 2015 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.importer.transform;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.ExecuteWatchdog;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.output.CountingOutputStream;
+import org.apache.commons.lang.SystemUtils;
+import org.geoserver.data.util.IOUtils;
+import org.geoserver.importer.ImportData;
+import org.geoserver.importer.ImportTask;
+
+/**
+ * Generic file translator getting a set of options, an input file, and an output file
+ * 
+ * @author Andrea Aime - GeoSolutions
+ */
+public abstract class AbstractCommandLineTransform extends AbstractTransform implements
+        PreTransform {
+
+    private static final long serialVersionUID = 5998049960852782644L;
+
+    static final long DEFAULT_TIMEOUT = 60 * 60 * 1000; // one hour
+
+    List<String> options;
+
+    public AbstractCommandLineTransform(List<String> options) {
+        this.options = options;
+    }
+
+    /**
+     * @return the options
+     */
+    public List<String> getOptions() {
+        return options;
+    }
+
+    /**
+     * @param options the options to set
+     */
+    public void setOptions(List<String> options) {
+        this.options = options;
+    }
+
+    @Override
+    public boolean stopOnError(Exception e) {
+        return true;
+    }
+
+    @Override
+    public void apply(ImportTask task, ImportData data) throws Exception {
+        boolean inline = isInline();
+        File executable = getExecutable();
+        File inputFile = getInputFile(data);
+        Map<String, File> substitutions = new HashMap<>();
+        substitutions.put("input", inputFile);
+        File outputDirectory = null;
+        File outputFile = null;
+        if (!inline) {
+            outputDirectory = inline ? null : getOutputDirectory(data);
```

Comment:
since we are inside if(!inline) the inline ? null : getOutputDirectory(data) can be replaced by getOutputDirectory()